src/tcp_connections.rs

```diff
@@ -46,31 +45,43 @@ pub fn upgrade_tcp(stream: TcpStream) -> io::Result<(TcpStream, Sender<WriteEven
 }
 
 fn upgrade_writer(mut stream: TcpStream) -> Sender<WriteEvent> {
+    use bincode::SizeLimit;
+    use maidsafe_utilities::serialisation::serialise_with_limit;
+
     let (tx, rx) = mpsc::channel();
-    let _ = unwrap_result!(thread::Builder::new()
-                               .name("TCP writer".to_owned())
-                               .spawn(move || {
-                                   while let Ok(event) = rx.recv() {
-                                       match event {
-                                           WriteEvent::Write(data) => {
-                                               use std::io::Write;
-                                               let msg = unwrap_result!(serialise(&data));
-                                               if stream.write_all(&msg).is_err() {
-                                                   break;
-                                               }
-                                           }
-                                           WriteEvent::Shutdown => break,
-                                       }
-                                   }
-                                   stream.shutdown(Shutdown::Both)
-                               }));
+    let _ = thread!("TCP writer", move || {
+        while let Ok(event) = rx.recv() {
+            match event {
+                WriteEvent::Write(data) => {
+                    use std::io::Write;
+                    use byteorder::{WriteBytesExt, LittleEndian};
+
+                    let payload = unwrap_result!(serialise_with_limit(&data, SizeLimit::Infinite));
+                    let size = payload.len() as u32;
+                    let mut little_endian_size_bytes = Vec::with_capacity(4);
```

Comment:
Ya that would be nice - i had used that initially, but seems that the trait is not implemented for `[u8; 4]`, so the next thing to do was to use a vec. Array will give a compile error:
```src/tcp_connections.rs:67:61: 67:70 error: no method named `write_u32` found for type `[u8; 4]` in the current scope
src/tcp_connections.rs:67                     unwrap_result!(little_endian_size_bytes.write_u32::<LittleEndian>(size));
                                                                                      ^~~~~~~~~
src/tcp_connections.rs:67:21: 67:94 note: in this expansion of unwrap_result! (defined in <maidsafe_utilities macros>)
<maidsafe_utilities macros>:2:1: 4:28 note: in this expansion of unwrap_result! (defined in <maidsafe_utilities macros>)
src/tcp_connections.rs:57:13: 82:7 note: in this expansion of thread! (defined in <maidsafe_utilities macros>)
src/tcp_connections.rs:67:61: 67:70 note: the method `write_u32` exists but the following trait bounds were not satisfied: `[u8; 4] : std::io::Write`, `[u8] : std::io::Write`
error: aborting due to previous error
Could not compile `crust`.
```